### PR TITLE
NcSelect: make same height and color as NcTextField

### PR DESCRIPTION
**Pull request by GVodyanov**
### ☑️ Resolves

- Fix https://github.com/nextcloud/calendar/issues/6155

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
image | image

### CODE REVIEW

**Review by marcoambrosini — CHANGES_REQUESTED** [review]
Hi @GVodyanov, the height of these elements needs to be --default-clickable-area, the input fields will also shrink a bit soon

**Comment by GVodyanov** [comment]
> Hi @GVodyanov, the height of these elements needs to be --default-clickable-area, the input fields will also shrink a bit soon

That's what it already should be set to actually

File: src/components/NcSelect/NcSelect.vue [hunk 1]
```diff
@@ -1263,12 +1265,14 @@ body {
 
 	.vs__selected-options {
 		// If search is hidden, ensure that the height of the search is the same
-		min-height: 40px; // 36px search height + 4px search margin
+		min-height: auto; // 36px search height + 4px search margin
```
Comment (marcoambrosini) [inline src/components/NcSelect/NcSelect.vue:1268]: So this min-height is unrelated?

File: src/components/NcSelect/NcSelect.vue [hunk 2]
```diff
@@ -1186,6 +1186,7 @@ body {
 
 	.vs__selected {
 		height: 32px;
+		margin: 0;
```
Comment (susnux) [inline src/components/NcSelect/NcSelect.vue:1189]: This causes the selected elements to directly "dock" onto the top border of the select.

**Review by susnux — CHANGES_REQUESTED** [review]
The color changes should only be applied to the placeholder, not the input element as the input text must stay main text color.

File: src/components/NcSelect/NcSelect.vue (hunk 1 above)
Comment (GVodyanov) [reply src/components/NcSelect/NcSelect.vue:1268]: Yeah, that's the min-height of the dropdown with the options

File: src/components/NcSelect/NcSelect.vue (hunk 2 above)
Comment (GVodyanov) [reply src/components/NcSelect/NcSelect.vue:1190]: Made it the same height so it doesn't matter now

**Comment by GVodyanov** [comment]
I made the height of everything `--default-clickable-area` all the way through, does it seem ok?

**Review by marcoambrosini — APPROVED** [review]

File: src/components/NcSelect/NcSelect.vue [hunk 3]
```diff
@@ -1194,15 +1195,19 @@ body {
...
 	}
 
 	.vs__search, .vs__search:focus {
-		margin: 2px 0 0;
+		margin: 0;
 	}
 
 	.vs__dropdown-toggle {
 		position: relative;
-		max-height: 100px;
```
Comment (susnux) [inline src/components/NcSelect/NcSelect.vue:1205]: Why this change? This can lead to really long element instead of scrolling. (like if you select many tags or something)

File: src/components/NcSelect/NcSelect.vue [hunk 4]
```diff
@@ -1262,13 +1267,14 @@ body {
 	}
 
 	.vs__selected-options {
-		// If search is hidden, ensure that the height of the search is the same
```
Comment (susnux) [inline src/components/NcSelect/NcSelect.vue:1265]: Why removing the comment?

Comment (GVodyanov) [reply src/components/NcSelect/NcSelect.vue:1265]: Typo

File: src/components/NcSelect/NcSelect.vue (hunk 3 above)
Comment (GVodyanov) [reply src/components/NcSelect/NcSelect.vue:1205]: Ah ok

Comment (susnux) [reply src/components/NcSelect/NcSelect.vue]: This is changing the behavior a lot:

before | after
---|---
Screenshot 2024-07-22 at 15-25-07 Nextcloud Vue Style Guide|Screenshot 2024-07-22 at 15-24-58 Nextcloud Vue Style Guide

File: src/components/NcSelect/NcSelect.vue (hunk 2 above)
Comment (susnux) [reply src/components/NcSelect/NcSelect.vue:1189]: It does now every selected element has no gap to the others:

before | after
---|---
Screenshot 2024-07-22 at 15-23-35 Nextcloud Vue Style Guide|Screenshot 2024-07-22 at 15-23-31 Nextcloud Vue Style Guide
Screenshot 2024-07-22 at 15-23-06 Nextcloud Vue Style Guide|Screenshot 2024-07-22 at 15-22-59 Nextcloud Vue Style Guide

Comment (susnux) [reply src/components/NcSelect/NcSelect.vue:1189]: So easiest: Just add back like 2px margin and make the height `--default-clickable-area - 2 * margin-size`.
For example:
```css
height: calc(var(--default-clickable-area) - var(--default-grid-baseline));
margin: calc(var(--default-grid-baseline) / 2);
```

Comment (GVodyanov) [reply src/components/NcSelect/NcSelect.vue]: Yep, added it!

**Review by susnux — APPROVED** [review]

**Comment by susnux** [comment]
Please remember to add correct labels:
1. State of the PR (e.g. `3. to review`)
2. But more important the type of the PR: `bug`, `enhancement` or `refactor` this is needed for changelog generation.

Also always remember to backport to *next* (there is a TODO section in the PR template for this :) ).

File: src/components/NcSelect/NcSelect.vue [hunk 5]
```diff
@@ -1263,12 +1269,13 @@ body {
 
 	.vs__selected-options {
 		// If search is hidden, ensure that the height of the search is the same
-		min-height: 40px; // 36px search height + 4px search margin
+		min-height: var(--default-clickable-area);
```
Comment (Antreesy) [inline src/components/NcSelect/NcSelect.vue:1272]: Selector specificity either isn't big enough or scoped enough here (see leaked from server and viewer):
image

Even with that applied, NcSelect height is 36px, not 34px

Comment (susnux) [reply src/components/NcSelect/NcSelect.vue:1272]: There is #5883
